main.md

```diff
@@ -193,180 +199,327 @@ Note: Applications MUST ensure that their authorization data types do not collid
          ]
       },
-      { 
-         "path":"/myfiles/A/X",
-         "access":[ 
+      {
+         "path": "/myfiles/A/X",
+         "access": [
             "read",
             "write"
          ]
       }
    ]
 }
 ```
+## Relationship to "scope" parameter {#scope}
+
+`authorization_details` and `scope` can be used in the same authorization request for carrying independent authorization requirements. 
+
+The AS MUST keep both requirements sets separate when reflecting this data to the client and passing this data through to resource servers.
+
+For user convenience, the AS is supposed to merge the requirements when asking the user for consent.
+
+### Scope value "openid" and "claims" parameter
+
+OpenID Connect [@OIDC] specifies the JSON-based `claims` request parameter that can be used to specify the claims a client (acting as OpenID Connect Relying Party) wishes to receive in a fine-grained and privacy preserving way as well as assign those claims to a certain delivery mechanisms, i.e. ID Token or userinfo response. 
+
+The combination of the scope value `openid` and the additional parameter `claims` can be used beside `authorization_details` in the same way as every other non-OIDC scope value. 
+
+Alternatively, there could be an authorization data type for OpenID Connect. (#openid) gives an example of how such an authorization data type could look like.
+
+## Relationship to "resource" parameter
+
+The request parameter `resource` [@I-D.ietf-oauth-resource-indicators] indicates to the AS the resource(s) where the client intends to use the access tokens issued based on a certain grant. This mechanism is a way to audience-restrict access tokens and to allow the AS to create resource server specific access tokens. 
+
+If a client uses `authorization_details` with `locations` elements in an authorization request, it MAY omit the `resource` parameter since the information conveyed in the `authorization_details` object are a superset of what can be conveyed in the `resource` parameter.
+
+Clients requiring audience restricted access tokens are RECOMMENDED to use the `resource` parameter in token requests to allow the AS to narrow down the privileges of the access token to specific permissions for individual operations on specific resource servers (see [@I-D.ietf-oauth-security-topics], Section 3.3). 
+
+If the client used `authorization_details` with `locations` elements in the authorization request, the AS MUST utilize this data to filter the authorization data objects applicable to the respective `resource`. This process will use the `resource` string as prefix to filter the `locations` elements of the authorization details elements.
+
+Given the example in (#authz_details), a client could request an access token using a `resource` value of `https://example.com/payments`, which would produce an access token containing the `payment_initation` but not the `account_information` authorization details as shown in the following example:
```

Comment:
I read this several times before understanding it (and not sure I do now). But maybe clarify that the example in example in (#authz_details) is the authorization request and a you are (presumably?) talking about a subsequent access token request with a refresh token and the resultant access token would contain or reference the JSON below as some claim value. If I even understood that all correctly...

Reply:
Also I'm not a huge fan of having this place for application/API specific authz details but that it might have some of these predefined data elements types and one of those elements has specific behavior relative to a request parameter in the token request.

Reply:
I tried to improve the text. Please review.

Reply:
I'm honestly having a hard time following all the threads and comments and updates but I do think the changes in 5626d118e29e465c57c7af7f0ce33f80d75dd3d3 improve it.

Reply:
I would suggest to merge the PR after we have came to a conclusion regarding scope/authz details merge and review the result again.